### DISCUSSION TITLE
Proposal to share each node storage to get a decentalized IPFS solution

### DISCUSSION THREAD

**Discussion by carry2web** [deso-protocol/dips#249] [Large CIPs]
I propose to start adding IPFS gateway(s) and IPFS service to each deso node to make their storage available as a decentralized storage solution for all deso images/videos content.
https://docs.ipfs.tech/install/

Currently those images and videos are stored on a Google workspace drive and CloudFlare Stream afaik.

With this proposal you will contribute to the decentralized nature AND save costs,

I like to hear your thoughts on this.
Related Q&A: https://github.com/deso-protocol/dips/discussions/179

@lazynina @diamondhands0

**Comment by StarGeezerPhil** [comment]
All for this if it's feasible and could be integrated into the backend - meaning all nodes could contribute to a fully decentralised file storage solution.

Possibly active for those nodes running archival mode or a separate IPFS_ACTIVE flag.

If optional, a carrot for operators to support the service could be them having access to upload via their own upload API.

**Comment by carry2web** [comment]
Discussion on-chain: https://desocialworld.com/posts/c1382b87ef0f97aa6901305d04af277a19b34816789694defdaa245cbe168b62

**Reply by carry2web** [reply]
@lazynina @diamondhands0 are you willing to comment? Is Deso Drive your answer? When to expect?

**Comment by carry2web** [comment]
Got a response from @mosster on-chain https://desocialworld.com/posts/bfd3b662628969ab17f7fbf47f2d6d2497dd454648f75ceb67df4a2946e55000
"
What [@SafetyNet](https://desocialworld.com/u/SafetyNet) is proposing is to standardize IPFS into the DeSo Nodes. Note that, IPFS is also not a great solution, your content can be deleted at any time if storage costs are not paid for - versus Arweave which is permanent.

IPFS gateways can also be censored: [torrentfreak.com/cloudflare-disables-access-to-pirated-content-on-its-ipfs-gateway-230324](https://torrentfreak.com/cloudflare-disables-access-to-pirated-content-on-its-ipfs-gateway-230324/)

This is why we let applications make their own assumptions for now whether they want to use CF, S3, IPFS, Storj, Arweave — until we upgrade to DeSo Drive, a more integrated solution later. This is also how Lens and Farcaster approach it."

**Reply by ArnoudvanderPlas** [reply]
Let me explain my statement: "Half truths". I think there is more than one way to approach the storage solution to begin with, all with their up- and downsides. To my knowledge, it is possible to securely store our data using IPFS or an IPFS related/based system like file coin.https://spheron.medium.com/ipfs-a-censorship-resistant-file-sharing-system-6fe46f8cb22f 
To say it is not great, censorship resistant or being at risk of being lost are not entirely or necessarily true. It greatly depends on the implementation of said solutions and the distribution of nodes and data IMHO. Arweave is also a great avenue to explore. It brings permanency but also higher costs. On the whole, we as a community have some exploring to do. At this point, @safetynet has put out an extensive list of different avenues of approach to the storage issue. IPFS is on this list, nothing more. I think It would not be entirely correct to say we are proposing to standardize that specific solution. All in all, my statement was meant to say we should not be dismissive of the proposed solutions based on a single point of view. In my opinion, free choice of storage solution(s) is a great thing to have. Separately, it could be advantageous to have a universal standard storage solution. The advantage being it providing more seamless services?

**Comment by carry2web** [comment]
We'll discuss possible storage in a SafetyNet meeting on vibehut shortly. Feel free to join.

**Comment by ArnoudvanderPlas** [comment]
I know that and they are half truths

Met vriendelijke groet,
*Arnoud van der Plas*

telefoon: +31625363229
email: ***@***.***




Op zo 15 dec 2024 om 01:04 schreef Carry Megens ***@***.***>
